stock_lot_production_date/models/stock_move_line.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _assign_production_lot(self, lot):
+        super()._assign_production_lot(lot)
+        if lot.product_id.use_production_date:
+            lot.production_date = fields.Datetime.now()
```

Comment:
I still don't know if this auto-assignment should be done only when validating a MO or not.
With this implementation, the production date will be set even when we create a lot through a normal stock transfer.

Reply:
This part has been removed and will be handled by https://github.com/OCA/manufacture/pull/934